fakestorage/server.go

```diff
@@ -211,9 +212,22 @@ func newServer(options Options) (*Server, error) {
 	return &s, nil
 }
 
+func unescapeMuxVars(vars map[string]string) map[string]string {
+	m := make(map[string]string)
+	for k, v := range vars {
+		r, err := url.PathUnescape(v)
+		if err == nil {
+			m[k] = r
+		} else {
+			m[k] = v
+		}
+	}
+	return m
+}
+
 func (s *Server) buildMuxer() {
 	const apiPrefix = "/storage/v1"
-	s.mux = mux.NewRouter()
+	s.mux = mux.NewRouter().SkipClean(true).UseEncodedPath()
```

Comment:
Reason for this.

> UseEncodedPath tells the router to match the encoded original path to the routes. For eg. "/path/foo%2Fbar/to" will match the path "/path/{var}/to".

This fix issue, that if we use names like `/file.txt`, in this case it will be `/download/storage/v1/b/{bucketName}/o/%2Ffile.txt` in url path, which mux trying to fix by redirect to `/download/storage/v1/b/{bucketName}/o/file.txt`, which is incorrect behaviour for GCP